Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -371,13 +373,12 @@ WarpX::ApplyInverseVolumeScalingToCurrentDensity (MultiFab* Jx, MultiFab* Jy, Mu
             Jr_arr(i,j,0,0) /= (2.*MathConst::pi*r);
 
             for (int imode=1 ; imode < nmodes ; imode++) {
-                const Real ifact = ( (imode%2) == 0 ? +1. : -1.);
                 // Wrap the current density deposited in the guard cells around
                 // to the cells above the axis.
                 // Note that Jr(i==0) is at 1/2 dr.
                 if (rmin == 0. && 0 <= i && i < ngJ) {
-                    Jr_arr(i,j,0,2*imode-1) -= ifact*Jr_arr(-1-i,j,0,2*imode-1);
-                    Jr_arr(i,j,0,2*imode) -= ifact*Jr_arr(-1-i,j,0,2*imode);
+                    Jr_arr(i,j,0,2*imode-1) -= Jr_arr(-1-i,j,0,2*imode-1);
+                    Jr_arr(i,j,0,2*imode) -= Jr_arr(-1-i,j,0,2*imode);
```

Comment:
Hi @dpgrote ! Thanks for these changes.
I am trying to understand them and to use RZ (as it is much faster than 3D!).

Why do you need the two lines? if you are looking over the modes, wouldn't one of them suffice?

Reply:
Jr holds complex data with the real and imaginary split apart. The (2*imode-1) is the real part and (2*imode) is the imaginary part. Also note that for mode zero, there is only the real part.

Reply:
Thanks @dpgrote ! It is more clear now how you fixed this.